tests/ducktape/consumer_strategy.py

```diff
@@ -118,50 +283,91 @@ def poll_messages(self, topic_name, test_duration, start_time, consumed_containe
                 poll_end = time.time()
 
                 poll_latency_ms = (poll_end - poll_start) * 1000
-                poll_times.append(poll_latency_ms)
 
                 if self.metrics:
                     self.metrics.record_api_call(poll_latency_ms)
 
                 if msg is None:
-                    # Timeout - no message received
                     if self.metrics:
                         self.metrics.record_timeout()
                     continue
 
                 if msg.error():
-                    # Error occurred
                     if self.metrics:
                         self.metrics.record_error(str(msg.error()))
                     self.logger.error(f"Consumer error: {msg.error()}")
                     continue
 
-                # Process the single message
+                # Deserialize message
+                msg = self._deserialize_message(msg, key_deserializer, value_deserializer, topic_name, messages_consumed)
                 consumed_container.append(msg)
                 messages_consumed += 1
+                self._record_message_metrics(msg, poll_latency_ms)
 
-                if self.metrics:
-                    self.metrics.record_processed_message(
-                        message_size=len(msg.value()) if msg.value() else 0,
-                        topic=msg.topic(),
-                        partition=msg.partition(),
-                        offset=msg.offset(),
-                        operation_latency_ms=poll_latency_ms
-                    )
-
-                # Progress tracking (removed verbose logging)
-
-        finally:
             consumer.close()
-
-        return messages_consumed
+            return messages_consumed
+        except Exception:
+            consumer.close()
+            raise
 
 
 class AsyncConsumerStrategy(ConsumerStrategy):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._consumer_instance = None
 
+    async def build_deserializers(self, deserialization_type):
```

Comment:
This seems like a helper function the core library should offer -- might be worth a follow-up PR considering exposing this closer to the SR client?

Reply:
+1, we should spend some time extracting those common util functionalities to core lib